bot/client.go

```diff
@@ -0,0 +1,64 @@
+package bot
+
+import (
+	"errors"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/nats-io/nats.go"
+	"github.com/rs/zerolog/log"
+
+	"github.com/domino14/macondo/config"
+	pb "github.com/domino14/macondo/gen/api/proto/macondo"
+	"github.com/domino14/macondo/move"
+	"github.com/domino14/macondo/runner"
+)
+
+type Client struct {
+	// NATS connection
+	nc      *nats.Conn
+	channel string
+}
+
+func MakeRequest(game *runner.GameRunner, config *config.Config) ([]byte, error) {
+	history := game.History()
+	if history.Variant == "" {
+		history.Variant = "CrosswordGame"
+	}
+	if history.Lexicon == "" {
+		history.Lexicon = config.DefaultLexicon
+	}
+	req := pb.BotRequest{GameHistory: history}
+	return proto.Marshal(&req)
+}
+
+// Send a game to the bot and get a move back.
+func (c *Client) RequestMove(game *runner.GameRunner, config *config.Config) (*move.Move, error) {
+	data, err := MakeRequest(game, config)
+	if err != nil {
+		return nil, err
+	}
+	res, err := c.nc.Request(c.channel, data, 10*time.Second)
```

Comment:
Once we add a simming bot, this timeout might be a bit short. We'll have to think about how to handle that.

Reply:
i think we will eventually have to have an async request/response, rather than waiting with a timeout, but that is its own set of complications so it's a lot easier to develop the bot with the synchronous response.

Reply:
I think keeping it synchronous is probably ok actually as long as we can ensure this runs in its own goroutine. One thought I had was to limit the timeout to something like a few minutes. But we would also have an additional timeout (with a `context.Context`) on a per-play (or a per-sim basis) that is shorter than the NATS timeout. When that context times out, the simulator or endgame or whatever time-consuming thing can be signaled to clean up and return its best value (if it hasn't already returned). In general in Go it's totally ok to do blocking synchronous functions as long as you run them inside goroutines, and it might actually be preferred a lot of the time (makes reasoning about the code easier!)